### PR TITLE
Auth0 Token usage and RBAC

### PR DESCRIPTION
**Pull request by bseptember**
### Auth0 Token usage and Role Base Access Control

**Overview**:

Use token from auth0 instead of getting it from the backend. From this token we receive after sign in, we validate whether it has correct roles/permissions assigned to the user which is set in the dashboard of Auth0.

**Note**:

This branch needs the server RGO branch with the same name to work.

**Future**:

Received from auth0 is the id token which contains the social connection user profile picture link which can be used as the profile picture in our application.

**Known issues**:

When the user is signed in with Auth0, but the backend server is offline, then the app still gets routed to other pages. checkBackendConnection can be used along with other changes to fix this issue.

### CODE REVIEW

File: src/app/ats-guard.guard.ts [hunk 1]
```diff
@@ -12,23 +11,17 @@ export class AtsPageGuard {
   constructor(private navService: NavService,
     private authService: AuthService,
     private cookieService: CookieService,
-    private authAccessService: AuthAccessService,
   ) { }
 
   canActivate(): any {
-    if (this.navService.isHris == false) {
+    if (false == this.navService.isHris) {
```
Comment (tbddorothy) [inline src/app/ats-guard.guard.ts:17]: What's the point of this switch over?

File: src/app/components/hris/sign-in/sign-in.component.ts [hunk 2]
```diff
@@ -55,77 +38,85 @@ export class SignInComponent {
...
+
+        this.authService.store.dispatch(GetLogin({ payload: userData }));
+        if (window.innerWidth > 776)
+          this.navService.showNavbar = true;
+        else {
+          this.navService.showSideBar = true;
+        }
+
+        this.sharedPropprtyAccessService.setAccessProperties();
+        // TODO: put back in
```
Comment (tbddorothy) [inline src/app/components/hris/sign-in/sign-in.component.ts:95]:
```suggestion
        // TODO: put back in once ATS is available
```

File: src/app/hris-guard.guard.ts [hunk 3]
```diff
@@ -13,22 +11,17 @@ export class HrisPageGuard {
   constructor(private navService: NavService,
     private authService: AuthService,
     private cookieService: CookieService,
-    private authAccessService: AuthAccessService
   ) { }
 
   canActivate(): any {
-    if (this.navService.isHris) {
+    if (true == this.navService.isHris) {
```
Comment (tbddorothy) [inline src/app/hris-guard.guard.ts:17]: Ditto here aswell

File: src/app/services/shared-services/auth-access/auth.service.ts [hunk 4]
```diff
@@ -1,69 +1,151 @@
...
+      tap(isAuthenticated => {
+        if (!isAuthenticated) {
+          throw new Error('User not authenticated');
+        }
+      })
+    ));
+
+    const accessToken = await firstValueFrom(this.auth0.getAccessTokenSilently().pipe(take(1)));
+    if (accessToken) {
+      console.log("Extracted access token:", accessToken);
```
Comment (tbddorothy) [inline src/app/services/shared-services/auth-access/auth.service.ts:74]: Please remove all console logs

File: src/app/services/shared-services/auth-access/auth.service.ts [hunk 5]
```diff
@@ -1,69 +1,151 @@
...
+      const decodedAccessToken = this.decodeJwt(accessToken);
+      console.log("json access token:", decodedAccessToken);
```
Comment (tbddorothy) [inline src/app/services/shared-services/auth-access/auth.service.ts:76]: Ditto

File: src/app/services/shared-services/auth-access/auth.service.ts [hunk 6]
```diff
@@ -1,69 +1,151 @@
...
+      const permissions = decodedAccessToken.permissions || [];
+      console.log("Permissions:", permissions);
```
Comment (tbddorothy) [inline src/app/services/shared-services/auth-access/auth.service.ts:78]: Ditto

File: src/app/services/shared-services/auth-access/auth.service.ts [hunk 7]
```diff
@@ -1,69 +1,151 @@
...
+      tap(isAuthenticated => {
+        if (!isAuthenticated) {
+          throw new Error('User not authenticated.');
+        }
+      })
+    ));
+
+    const userInfo = await firstValueFrom(this.auth0.user$.pipe(take(1)));
+    if (userInfo) {
+      console.log("userinfo",userInfo);
```
Comment (tbddorothy) [inline src/app/services/shared-services/auth-access/auth.service.ts:106]: Ditto

File: src/app/ats-guard.guard.ts (hunk 1 above)
Comment (bseptember) [reply src/app/ats-guard.guard.ts:17]: If someone accidently removes the equal sign then it becomes a variable assignment and not a comparison. This bug will be difficult to find. However I do note that Angular uses "===" for assignments and if a dev. removes an equal then the compiler does detect it. Unlike in c#.

**Review by dvanhout89 — DISMISSED** [review]

**Review by EduanO — DISMISSED** [review]

**Review by tbddorothy — APPROVED** [review]

File: src/app/components/hris/sign-in/sign-in.component.ts [hunk 8]
```diff
@@ -68,66 +69,77 @@ export class SignInComponent {
...
           else
             this.navService.showSideBar = true;
-
+  
           this.sharedPropprtyAccessService.setAccessProperties();
           this.initialUserNavigation();
         },
+        error: (err) => {
+          console.error("Login process error:", err);
+        }
```
Comment (Juanro-Minne) [inline src/app/components/hris/sign-in/sign-in.component.ts:136]: should we maybe use a snackbar for this or is the console error fine?

**Review by ReinoRabbit — APPROVED** [review]

**Review by Juanro-Minne — APPROVED** [review]

**Review by EduanO — APPROVED** [review]

**Pull request merged**